lib/diagnostic.rb

```diff
@@ -1,10 +1,12 @@
 # Ruby enumerable diagnostic
 class Diagnostic
+
   # Question 1
   # Write ruby code to return the longest entry in the
   # enumerable supplied.  Assume each entry implements a length
   # method that returns a Fixnum.
   def longest(enum)
+    enum.reduce(0) { |m, o| m = m.length < o.length? o : m }
```

Comment:
Enumerable provides some simpler methods for doing this kind of work: `max` and `max_by`.
e.g.
`enum.max { |a, b| a.length <=> b.length }`
`enum.max_by { |entry| entry.length }`